### PR TITLE
Sort listings frontend

### PR DESCRIPTION
**Pull request by rtman**
### Description:

This is the front end changes for the sort listings update #2414 . I have followed the zeplin screens that @micahalcorn linked in the [issue](https://github.com/OriginProtocol/origin/issues/2414). 

Marketplace tests are intermittently failing, I don't think that is related to this change though.

Currently there isn't a test build as the backend changes haven't been deployed, but I thought since I won't be contributing as much I should put this up.

### Checklist:

- [✅] Test your work and double-check to confirm that you didn't break anything
- [✅] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [✅] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)

### CODE REVIEW

**Comment by franckc** [comment]
Thanks @rtman !

@micahalcorn do you mind doing a quick code review ?
On my side I will test on my local and can take care of any changes you recommend.

File: dapps/marketplace/src/pages/listings/Listings.js [hunk 1]
```diff
@@ -294,6 +323,19 @@ export default withGrowthRewards(
...
 
 require('react-styl')(`
+  .listings-menu-bar 
+    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
+    padding: 0 1rem;
+    position: relative;
+    display: flex;
+    flex-wrap: wrap;
+    align-items: center;
+    justify-content: space-between;
```
Comment (micahalcorn) [inline dapps/marketplace/src/pages/listings/Listings.js]: I think the internal preference is for no semicolons these days. Not sure if this is covered by the linter 😐

Comment (franckc) [reply dapps/marketplace/src/pages/listings/Listings.js]: Cleaned this up.

**Comment by franckc** [comment]
Tested functionality on my local.
Tweaked a bit the UI:
 - Dismiss the Sort By modal on mobile as soon as a selection is made
 - Removed scroll bar on search dropdown showing the featured categories

Here is how it looks currently:
**Mobile**
Sort By:
MobileSortBy

Featured categories:
MobileFeaturedCategories

**Desktop**
Sort By dropdown:
DesktopSortBy

Featured categories dropdown:
DesktopFeaturedCategories

**Comment by franckc** [comment]
@micahalcorn @nick This is ready for review.

**Comment by franckc** [comment]
Fixed the following issues found during previous round of review:
- Click anywhere should dismiss sort by dropdown
- Search bar not centered
- Add sort+order params to URL

The back-end fix for the sorting being broken is in separate PR #2985

**Comment by franckc** [comment]
Back-end fix deployed to dev+staging.

@micahalcorn Here is a build to test: https://originprotocol.github.io/test-builds/franck-sort-fe/#/

**Comment by franckc** [comment]
Another confusing thing I noticed about our current search UI is that once a category is selected, it is hard to reset. The only way I found was to go to home page. Not sure if this was the intended UX ? This should be addressed as a separate PR though, since it is not related to sorting.

**Comment by micahalcorn** [comment]
Great, thx @franckc 👏 We'll make you a front-end expert after all.

**Comment by micahalcorn** [comment]
Yeah @franckc 🤔 Clearing the search query is also problematic on mobile. We are working on some design improvements.

**Pull request merged**